src/app/SubscriptionResumptionSessionEstablisher.cpp

```diff
@@ -36,6 +36,32 @@ class AutoDeleteEstablisher
     SubscriptionResumptionSessionEstablisher * mSessionEstablisher;
 };
 
+struct SubscriptionResumptionSessionEstablishRecord
+{
+    chip::SubscriptionId mSubscriptionId;
+    uint16_t mSessionEstablishRetries = UINT16_MAX;
+};
+
+static SubscriptionResumptionSessionEstablishRecord sSubscriptionResumptionRecords[CHIP_IM_MAX_NUM_SUBSCRIPTIONS];
```

Comment:
This should go in `SimpleSubscriptionResumptionStorage` and not in the establisher. 
This array isn't manage correctly. If a subscription entry is removed, this data structure is not updated.

Reply:
`SimpleSubscriptionResumptionStorage` is only an example implement of `SubscriptionResumptionStorage`. Should we add the retrying logic in it?

Added updating the array when entry is removed.

Reply:
Also I found another cause why the sever app does not retry on resuming subscriptions.

The previous work flow is:
server app tries to resume subscription and create a read handler for it-> server app fails to establish the session-> the read handler is closed and `InteractionModelEngine::OnDone()` is called->sever app schedules another subscription resumption

But the current work flow is:
sever app tries to resume subscription->server app fails to establish the session

The read handler will not be created and `InterationModelEngine::OnDone()` will not be called. So server app will never schedule another subscription resumption.

Should we schedule subscription resumption when failing to establish the session? @mkardous-silabs @jtung-apple

Reply:
Subscription resumption should be scheduled when failing to establish the session. The scenario it covers is network outage. If wifi router get replaced / updated, etc, or if there's a general power outage for a day, the long-idle-time battery-powered device still should attempt to resume.

That's the reason the timeout-triggered subscription resumption is on its own timer, with the intention that that the power impact is not too much worse than regular wake+report.

After all, we want the user experience to be that after an outage of some sort, the house recovers over time without needing the user to run around the house and re-pair all the "lost" devices.

Reply:
I have already add the change of scheduling subscription resumption when failing to establish the session. Please take a review @mkardous-silabs @jtung-apple

Reply:
@wqx6 The intial comment was not addressed. The `SubscriptionInfo` structure is defined in the `src/app/SubscriptionResumptionStorage.h` class. Please ass this entry in that structure and not in a seperate structure.

Reply:
@mkardous-silabs Done, please take a look.